include/rocksdb/listener.h

```diff
@@ -233,6 +233,23 @@ struct CompactionJobInfo {
   CompactionJobStats stats;
 };
 
+struct SubcompactionJobInfo {
+  SubcompactionJobInfo() = default;
+
+  // the id of the column family where the compaction happened.
+  uint32_t cf_id;
+  // the name of the column family where the compaction happened.
+  std::string cf_name;
+  // the status indicating whether the compaction was successful or not.
+  Status status;
+  // the id of the thread that completed this compaction job.
+  uint64_t thread_id;
+  // the smallest input level of the compaction.
```

Comment:
Add job_id so we can use it to match the CompactionJobInfo later?

Reply:
It seems job id is only unique within each background thread, that's why I removed it.
```c++
// the job id, which is unique in the same thread.
  int job_id;
```

Reply:
We can put more thoughts on how to match subcompaction with main compaction later. Right now this PR covers our user case.